primedev/plugins/pluginmanager.h

```diff
@@ -0,0 +1,33 @@
+#ifndef PLUGIN_MANAGER_H
+#define PLUGIN_MANAGER_H
+
+#include <windows.h>
+
+class Plugin;
+
+class PluginManager
+{
+public:
+	const std::vector<Plugin>& GetLoadedPlugins() const;
+	const std::optional<Plugin> GetPlugin(HMODULE handle) const;
+	bool LoadPlugins(bool reloaded = false);
+	void LoadPlugin(fs::path path, bool reloaded = false);
+	void ReloadPlugins();
+	void RemovePlugin(HMODULE handle);
+
+	// callback triggers
+	void InformSqvmCreated(CSquirrelVM* sqvm) const;
+	void InformSqvmDestroying(CSquirrelVM* sqvm) const;
+	void InformDllLoad(HMODULE module, fs::path path) const;
+	void RunFrame() const;
+
+private:
+	void InformAllPluginsInitialized() const;
+
+	std::vector<Plugin> plugins;
+	std::string pluginPath;
```

Comment:
I'd prefer if member variables were prefixed with `m_`

Reply:
people wouldn't need to prefix member variables if they just used `this`